code/data_structures/stack/reverse_stack/reverse_stack.c

```diff
@@ -0,0 +1,137 @@
+// C program to reverse a stack using recursion
+#include<stdio.h>
+#include<stdlib.h>
+#define bool int
+ 
+/* structure of a stack node */
+struct sNode
+{
+    char data;
+    struct sNode *next;
+};
+ 
+/* Function Prototypes */
+void push(struct sNode** top_ref, int new_data);
+int pop(struct sNode** top_ref);
+bool isEmpty(struct sNode* top);
+void print(struct sNode* top);
+ 
+// Below is a recursive function that inserts an element
+// at the bottom of a stack.
+void insertAtBottom(struct sNode** top_ref, int item)
+{
+    if (isEmpty(*top_ref))
+        push(top_ref, item);
+    else
+    {
+ 
+        /* Hold all items in Function Call Stack until we
+           reach end of the stack. When the stack becomes
+           empty, the isEmpty(*top_ref)becomes true, the
+           above if part is executed and the item is inserted
+           at the bottom */
+        int temp = pop(top_ref);
+        insertAtBottom(top_ref, item);
+ 
+        /* Once the item is inserted at the bottom, push all
+           the items held in Function Call Stack */
+        push(top_ref, temp);
+    }
+}
+ 
+// Below is the function that reverses the given stack using
+// insertAtBottom()
+void reverse(struct sNode** top_ref)
+{
+    if (!isEmpty(*top_ref))
+    {
+        /* Hold all items in Function Call Stack until we
+           reach end of the stack */
+        int temp = pop(top_ref);
+        reverse(top_ref);
+ 
+        /* Insert all the items (held in Function Call Stack)
+           one by one from the bottom to top. Every item is
+           inserted at the bottom */
+        insertAtBottom(top_ref, temp);
+    }
+}
+ 
+/* Driveer program to test above functions */
+int main()
```

Comment:
Don't just stick `main` in the middle of your code, move it above the functions that are above it, and add function prototypes for them as well.